chainer/functions/connection/multihead_attention.py

```diff
@@ -0,0 +1,298 @@
+            :math:`S` is the source sequence length, :math:`B` is the
+            batch size, and :math:`E` is the embedding size.
+        expected_embedding_size (int): Total number of units of the model.
+        n_head (int): The number of parallel attention heads.
+        proj_in_W (:obj:`tuple`, :class:`~chainer.Variable` or :ref:`ndarray`):
+            Weight(s) to project `query`, `key`, and `value` vectors.
+            If the input sizes of `query`, `key`, and `value` are different,
+            this should be the tuple of three weights, otherwise, one weight.
+        proj_in_b (:class:`~chainer.Variable` or :ref:`ndarray`):
+            Bias added to projected `query`, `key`, and `value` vectors.
+        add_zero_attn (bool): If ``True``, add a new batch of zeros to
+            the key and value sequences at axis=1.
+        attention_dropout (float): Dropout ratio at the attention layer.
+        post_dropout (float): Dropout ratio at the output.
+        proj_out_W (:class:`~chainer.Variable` or :ref:`ndarray`):
+            Weight to project attention.
+        proj_out_b (:class:`~chainer.Variable` or :ref:`ndarray`):
+            Bias for projected attention.
+        key_padding_mask (:class:`~chainer.Variable` or :ref:`ndarray`):
+            If not ``None``, specified padding elements in the key
+            will be ignored by the attention.
+            The shape is :math:`(B, S)` where :math:`B` is the batch size,
+            and :math:`S` is the source sequence length.
+        attn_mask (:class:`~chainer.Variable` or :ref:`ndarray`):
+            Mask help attention ignores certain positions.
+            The shape is :math:`(L, L)` where :math:`L` is
+            the target sequence length.
+        dot_product_scaler: (float): Scaler for dot product. If ``None``,
+            :math:`1 / \\sqrt{embedding_size / n_head}` is used.
+        softmax_scaler (float): Softmax smoothing, or sharpening, coefficient.
+            This value is for cuDNN implementation.
+        return_weights (bool): If ``True``, return averaged attention weights.
+
+    Returns:
+        tuple: This function returns a tuple containing ``attn_output`` and
+        ``attn_output_weights``.
+
+        - ``attn_output`` is the output of attention whose shape is
+          :math:`(L, B, E)` where :math:`L` is the target sequence length,
+          :math:`B` is the batch size, and :math:`E` is the embedding size.
+        - ``attn_output_weights`` is the weights of attention whose shape is
+          :math:`(B, L, S)` where :math:`B` is the batch size,
+          :math:`L` is the target sequence length,
+          and :math:`S` is the source sequence length. If ``return_weights`` is
+          ``False``, this return value is ``None``.
+
+    .. seealso:: :class:`~chainer.links.MultiHeadAttention`
+
+    """
+    chainer.utils.experimental('chainer.functions.multihead_attention')
```

Comment:
Please add a statement `import chainer` on the top of this module.

Reply:
I think you meant to add this comment to chainer/links/connection/multihead_attention.py.